src/components/widgets/WebWidget.tsx

```diff
@@ -0,0 +1,39 @@
+import { useNavigation } from '@react-navigation/core';
+import React, { useCallback } from 'react';
+
+import { Icon, normalize, texts } from '../../config';
+import { ScreenName, WidgetProps } from '../../types';
+import { Image } from '../Image';
+
+import { DefaultWidget } from './DefaultWidget';
+
+export const WebWidget = ({ text, additionalProps }: WidgetProps) => {
+  const navigation = useNavigation();
+
+  const onPress = useCallback(() => {
+    navigation.navigate(ScreenName.Web, {
+      title: additionalProps?.staticContentTitle,
+      webUrl: additionalProps?.webUrl
+    });
+  }, [navigation, text]);
+
+  return (
+    <DefaultWidget
+      Icon={() =>
+        additionalProps?.image ? (
+          <Image
+            source={{ url: additionalProps?.image.url }}
+            style={{
+              height: normalize(additionalProps?.image?.height ?? 26),
+              width: normalize(additionalProps?.image?.width ?? 33)
+            }}
+          />
+        ) : (
+          <Icon.Url size={normalize(26)} />
+        )
+      }
+      onPress={onPress}
+      text={text as string}
```

Comment:
how come? `text?: string;` is already defined in `WidgetProps`:
maybe we need `export const WebWidget = ({ text = '', additionalProps }: WidgetProps) =>`?

Reply:
I got a warning about text being `undefined`, so I added this code, but it works with `text = ''`.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/de8d88b871253a95f408be4c21b30f50c9c0d382